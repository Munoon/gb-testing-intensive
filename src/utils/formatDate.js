export const formatDate = (date) => {
  return date.toDateString();
}