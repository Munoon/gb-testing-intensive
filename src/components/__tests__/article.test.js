import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import Article from '../article';
import '../../utils/formatDate'

const article = {
  title: "Title",
  publishedAt: "Mon Nov 23 2020 15:43:10 GMT+0300",
  summary: "long text",
  url: "/",
  imageUrl: "imageUrl",
  newsSite: 'some news',
};

jest.mock('../../utils/formatDate', () => ({
  formatDate: jest.fn(() => 'date')
}));


describe('article tests', () => {
  describe('snapshot tests', () => {
    it('matches snapshot with no article', () => {
      const component = render(
        <Article />
      );
  
      expect(component).toMatchSnapshot();
    });
  
    it('matches snapshot with article', () => {
      const component = render(<Article article={article} />);
  
      expect(component).toMatchSnapshot();
    });
  })

  describe('functionality tests', () => {
    it('renders placeholder when no article present', () => {
      const component = render(<Article />);
  
      const placeholder = component.getByText('SUMMARY').parentElement;
      expect(placeholder).toHaveClass('post-empty');
    });
  
    it("renders a link to source when article present", () => {
      const component = render(<Article article={article} />);
  
      const link = component.getByText(article.title).closest('a');
      expect(link).toHaveAttribute('href', article.url);
    });

    it("always passes", () => {
      render(<Article article={article} />);

      expect(1).toBe(1);
    });
  })
});
