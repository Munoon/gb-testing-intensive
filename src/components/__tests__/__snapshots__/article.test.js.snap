// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article tests matches snapshot with article 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="post"
        href="/"
      >
        <div
          class="header"
        >
          Title
        </div>
        <div
          class="imageWrapper"
        >
          <img
            alt="Title"
            class="image"
            src="imageUrl"
          />
        </div>
        <div
          class="summary"
        >
          long text
        </div>
        <div
          class="bottom"
        >
          <div
            class="source"
          >
            some news
          </div>
          <div
            class="date"
          >
            Mon Nov 23 2020
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="post"
      href="/"
    >
      <div
        class="header"
      >
        Title
      </div>
      <div
        class="imageWrapper"
      >
        <img
          alt="Title"
          class="image"
          src="imageUrl"
        />
      </div>
      <div
        class="summary"
      >
        long text
      </div>
      <div
        class="bottom"
      >
        <div
          class="source"
        >
          some news
        </div>
        <div
          class="date"
        >
          Mon Nov 23 2020
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`article tests matches snapshot with no article 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="post-empty"
      >
        <div
          class="header"
        >
          HEADER OF THE TITLE
        </div>
        <div
          class="image"
        >
          IMAGE WILL GO HERE
        </div>
        <div
          class="summary"
        >
          SUMMARY
        </div>
        <div
          class="bottom"
        >
          <div
            class="source"
          >
            SOURCE
          </div>
          <div
            class="date"
          >
            DATE
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="post-empty"
    >
      <div
        class="header"
      >
        HEADER OF THE TITLE
      </div>
      <div
        class="image"
      >
        IMAGE WILL GO HERE
      </div>
      <div
        class="summary"
      >
        SUMMARY
      </div>
      <div
        class="bottom"
      >
        <div
          class="source"
        >
          SOURCE
        </div>
        <div
          class="date"
        >
          DATE
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
